sunpy/net/vso/vso.py

```diff
@@ -168,6 +170,26 @@ def iter_errors(response):
         if not hasattr(prov_item, 'record') or not prov_item.record:
             yield prov_item
 
+def check_connection(url):
+    try:
+        if requests.get(url).status_code == 200:
+            return True
+        else:
+            return False
+    except (socket.error, socket.timeout) as e:
+        warnings.warn("Connection failed with error {}. \n Retrying with different url and port.".format(e))
+    
+def fallback_test(api,url,port):
+    if api is None:
+        if url is None or port is None:
+            for i in DEFAULT_URL_PORT:
+                url = i["url"]
```

Comment:
rather than defining these variables here just use `i['url']` wherever `url` is used at the moment.

Also change `i` to something descriptive like `mirror` or `endpoint`